pytorch_lightning/core/optimizer.py

```diff
@@ -50,6 +49,14 @@ def __init__(self,
 
         self.__dict__ = {k: v for k, v in optimizer.__dict__.items() if k != 'step'}
 
+        # save functions and attributes from optimizer
+        self.state_dict = optimizer.state_dict
+        self.load_state_dict = optimizer.load_state_dict
+        self.zero_grad = optimizer.zero_grad
+        self.add_param_group = optimizer.add_param_group
+        self.__setstate__ = optimizer.__setstate__
+        self.__getstate__ = optimizer.__getstate__
```

Comment:
Does that mean the line below `self.__class__ = type(...` doesn't work as we expect?
I thought it would handle that

Reply:
LARS in bolts was failing when this wasn't included. LARS in bolts is just a wrapper around an optimizer, where the optimizer within LARSWrapper sets these params as well

Reply:
I was making some test. It works, just attributes where missing.